src/mono/mono/mini/interp/interp.c

```diff
@@ -6390,6 +6390,24 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
 			mono_memory_barrier ();
 			MINT_IN_BREAK;
 		}
+		MINT_IN_CASE(MINT_MONO_EXCHANGE_I8)
```

Comment:
I think you need braces when declaring variables inside switch case because c++

Reply:
ah, that's could be why C++ is failing to build... Let me add a pair of braces.